actionpack/test/dispatch/session/test_session_test.rb

```diff
@@ -62,4 +62,9 @@ def test_fetch_returns_block_value
     session = ActionController::TestSession.new(one: "1")
     assert_equal(2, session.fetch("2") { |key| key.to_i })
   end
+
+  def test_session_id
+    session = ActionController::TestSession.new
+    assert_instance_of Rack::Session::SessionId, session.id
```

Comment:
```suggestion
    assert_instance_of String, session.id.public_id
```
I think this test shows better the problem of the previous implementation – `session.id.public_id` would fail with a `NoMethodError: undefined method `public_id' for "...":String`.